core/src/main/java/io/questdb/cutlass/http/HttpServer.java

```diff
@@ -392,7 +395,7 @@ public HttpConnectionContext newInstance(long fd, IODispatcher<HttpConnectionCon
 
         @Override
         public void done(HttpConnectionContext context) {
-            if (closed) {
+            if (closed || contextPool.get().currentSize() > connectionPoolMaxCapacity) {
```

Comment:
nit: we have the same potential problem in other places where we use `WeakObjectPool` and `WeakAutoClosableObjectPool`, e.g. `LineTcpConnectionContextFactory#LineTcpReceiver`. Does it make sense to store the max size in the pool classes themselves and modify `push()` method, so it gets `offer()` behavior?

Reply:
moved max size check into the pool

Reply:
Many thanks!